conn.go

```diff
@@ -196,39 +229,39 @@ func (c *conn) exec(ctx context.Context, query string, args []driver.Value) (dri
 	if err != nil {
 		return nil, err
 	}
-	body, err := c.doRequest(ctx, req)
+	body, err, _ := c.doRequest(ctx, req)
 	if body != nil {
 		body.Close()
 	}
 	return emptyResult, err
 }
 
-func (c *conn) doRequest(ctx context.Context, req *http.Request) (io.ReadCloser, error) {
+func (c *conn) doRequest(ctx context.Context, req *http.Request) (io.ReadCloser, error, bool) {
 	ctx, cancel := context.WithCancel(ctx)
 	transport := c.transport
 	c.cancel = cancel
 
 	if transport == nil {
 		c.cancel = nil
-		return nil, driver.ErrBadConn
+		return nil, driver.ErrBadConn, false
 	}
 
 	req = req.WithContext(ctx)
 	resp, err := transport.RoundTrip(req)
 	if err != nil {
 		c.cancel = nil
-		return nil, err
+		return nil, err, true
```

Comment:
You don't have to return the bool value. You can check error is a transport error. 
```
if _, ok := err.(Error);  !ok && err != driver.ErrBadConn {
```